### DISCUSSION TITLE
Create a Playground flet so you can try it online

### DISCUSSION THREAD

**Discussion by bobwatcherx** [flet-dev/flet#1234] [Ideas]
I have an idea so that flets can be tried and tested online with the Playground. Like the Flutter documentation, there's a Playground so we can test our code online without needing to Try it offline

**Comment by FeodorFitsner** [comment]
FletPad :) Nice idea! I think we can do that with Pyodide now.

**Comment by Muddassir-Farooq-official** [comment]
> FletPad :) Nice idea! I think we can do that with Pyodide now.

Check it out https://labdeck.com/python-designer/flet-gui-designer/

**Comment by MominIqbal-1234** [comment]
Hi @FeodorFitsner ,

I am building a web-based AI builder for Flet. As shown in the attached <img width="1918" height="906" alt="image" src="https://github.com/user-attachments/assets/fe3f9a9a-9570-4cb7-b2b4-cca9e19a201f" />
, I want to show a live UI preview inside the browser next to the code.

What is the best way to make this possible?

Run Flet directly in the browser using Pyodide (WASM)?

Run it on a backend server and display it in an iframe?

Manually generate HTML/CSS to visually mock the UI for the user demo?

Thanks for your help!

**Reply by FeodorFitsner** [reply]
Are you asking about https://flet.app (cause it's a screenshot from there)? Sorry, I didn't get the question.

**Reply by MominIqbal-1234** [reply]
Yes, actually I'm thinking about building a no-code platform for students, similar to FlutterFlow, but specifically for the Flet framework.

However, I'm not sure how to execute Flet code directly in the browser and display the GUI output. One option is to find a way to run Flet applications inside the browser. Another option is to create a drag-and-drop UI builder where every component the user adds automatically generates the corresponding Python Flet code. Users could then copy the generated code and run it directly on their own computers.

I think the idea has real potential, especially because there aren't many visual builders focused on Flet. The second approach (drag-and-drop UI that generates Flet code) is likely the better starting point because it's much easier to build and doesn't require running Python applications in the browser. Later, you could add features like live previews or cloud execution as the platform grows.

I'm not sure which approach is better. Do you think this is a good idea? What are your thoughts?
@FeodorFitsner

**Reply by FeodorFitsner** [reply]
But we've already built a no-code platform at https://flet.app/ that can be used by students! Let me know what features are you missing.
